### ISSUE TITLE
Deadlock on latest version

### ISSUE THREAD

**Issue by jaime29010** [DemonWav/MinecraftDevIntelliJ#25]
For some reason I get this issue while using the latest version, it renders the IDE unusable
The cause of it seems to be this plugin, but I'm not sure.
It just starts loading as normal, and when it finishes loading the project it freezes.
Removing the plugin fixes this issue.
[This appears](http://prntscr.com/bofnbb) at the bottom left part of the IDE
```
2016-07-04 01:56:13,219 [  10444]  ERROR - plication.impl.ApplicationImpl - Calling invokeAndWait from read-action leads to possible deadlock. 
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:127)
	at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:728)
	at com.demonwav.mcdev.util.Util.runWriteTask(Util.java:9)
	at com.demonwav.mcdev.platform.MinecraftModule.getInstance(MinecraftModule.java:68)
	at com.demonwav.mcdev.platform.MinecraftModule.getInstance(MinecraftModule.java:94)
	at com.demonwav.mcdev.platform.bukkit.BukkitFileIconProvider.getIcon(BukkitFileIconProvider.java:22)
	at com.intellij.util.IconUtil.getProvidersIcon(IconUtil.java:182)
	at com.intellij.util.IconUtil.access$200(IconUtil.java:49)
	at com.intellij.util.IconUtil$2.fun(IconUtil.java:145)
	at com.intellij.util.IconUtil$2.fun(IconUtil.java:136)
	at com.intellij.ui.DeferredIconImpl.evaluate(DeferredIconImpl.java:287)
	at com.intellij.ui.DeferredIconImpl$2$1$1.run(DeferredIconImpl.java:158)
	at com.intellij.ui.IconDeferrerImpl.evaluateDeferred(IconDeferrerImpl.java:122)
	at com.intellij.ui.DeferredIconImpl$2$1.run(DeferredIconImpl.java:155)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1178)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils$2.run(ProgressIndicatorUtils.java:80)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils$5.run(ProgressIndicatorUtils.java:141)
	at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:142)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:446)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:127)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:137)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:77)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:102)
	at com.intellij.ui.DeferredIconImpl$2.run(DeferredIconImpl.java:152)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:187)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

[Full log](https://github.com/DemonWav/MinecraftDevIntelliJ/files/345363/idea.txt)

**Comment by DemonWav** [comment]
Thank you, that gives me what I need to fix it. Sorry about that.

**Comment by theminecoder** [comment]
Can confirm. Was super confused at first as my logs never got that far for 90% of the time.

**Comment by DemonWav** [comment]
Please update, this should be fixed on alpha-0.0.19.

**Comment by theminecoder** [comment]
Yep seems to have disappeared! Thanks heaps
